src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -74,15 +73,22 @@ DkmClrValue IDkmClrResultProvider.GetClrValue(DkmSuccessEvaluationResult evaluat
 
         void IDkmClrResultProvider.GetChildren(DkmEvaluationResult evaluationResult, DkmWorkList workList, int initialRequestSize, DkmInspectionContext inspectionContext, DkmCompletionRoutine<DkmGetChildrenAsyncResult> completionRoutine)
         {
-            var dataItem = evaluationResult.GetDataItem<EvalResultDataItem>();
-            if (dataItem == null)
+            try
+            {
+                var dataItem = evaluationResult.GetDataItem<EvalResultDataItem>();
+                if (dataItem == null)
+                {
+                    // We don't know about this result.  Call next implementation
+                    evaluationResult.GetChildren(workList, initialRequestSize, inspectionContext, completionRoutine);
+                    return;
+                }
+                
+                completionRoutine(GetChildren(dataItem, workList, evaluationResult.StackFrame, initialRequestSize, inspectionContext));
```

Comment:
Shouldn't 'completionRoutine' be passed to GetChildren, to be called asynchronously? Same question for GetItems.